datafusion/core/src/dataframe/mod.rs

```diff
@@ -3337,52 +3337,6 @@ mod tests {
         Ok(())
     }
 
-    // Table 't1' self join
-    // Supplementary test of issue: https://github.com/apache/datafusion/issues/7790
```

Comment:
why is this test removed?

Reply:
Should have explained that more in the description. Before it tests that calling `with_column` after a self join without alias will fail in a certain way. After this change it will fail already at the self join. So there is no way to make it test what it testsed before and I belive the self join already has test coverage elsewhere.